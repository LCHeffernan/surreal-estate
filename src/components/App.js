import React, { useState } from "react";
import { Routes, Route } from "react-router-dom";
import "../styles/app.css";
import NavBar from "./NavBar";
import Properties from "./Properties";
import AddProperty from "./AddProperty";
import SavedProperties from "./SavedProperties";

const App = () => {
  const [userID, setUserID] = useState("");

  const handleLogin = (response) => {
    setUserID(response.id);
  };

  const handleLogout = () => {
    window.FB.logout(() => {
      setUserID("");
    });
  };

  return (
    <div className="App">
      <NavBar onLogin={handleLogin} onLogout={handleLogout} userID={userID} />
      <Routes>
        <Route path="/" element={<Properties userID={userID} />} />
        <Route path="add-property" element={<AddProperty />} />
        <Route
          path="saved-properties"
          element={<SavedProperties userID={userID} />}
        />
      </Routes>
    </div>
  );
};

export default App;
