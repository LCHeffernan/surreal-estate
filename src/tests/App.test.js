import React from "react";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import App from "../components/App";

describe("App", () => {
  xit("renders correctly", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  xit("renders learn react link", () => {
    render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );
    const linkElement = screen.getByText(/surreal estate/i);
    expect(linkElement).toBeInTheDocument();
  });
});
