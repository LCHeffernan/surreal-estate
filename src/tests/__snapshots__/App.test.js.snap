// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="navbar"
    >
      <img
        alt="logo of purple castle with words 'surreal estate'"
        class="logo"
        data-testid="logo"
        src="/logo.png"
      />
      <ul
        class="navbar-links"
      >
        <li
          class="navbar-link"
        >
          <a
            class="item"
            href="/"
          >
            View Properties
          </a>
        </li>
        <li
          class="navbar-link"
        >
          <a
            class="item"
            href="/add-property"
          >
            Add a Property
          </a>
        </li>
      </ul>
    </div>
    <div
      class="properties-container"
    >
      <div
        class="sidebar"
      >
        <div
          class="side-bar"
        >
          <div
            class="filter-city-title"
          >
            Filter by city
          </div>
          <div
            class="city-link-container"
          >
            <a
              class="city-link"
              href="/?query={\\"city\\":\\"Manchester\\"}"
            >
              Manchester
            </a>
          </div>
          <div
            class="city-link-container"
          >
            <a
              class="city-link"
              href="/?query={\\"city\\":\\"Leeds\\"}"
            >
              Leeds
            </a>
          </div>
          <div
            class="city-link-container"
          >
            <a
              class="city-link"
              href="/?query={\\"city\\":\\"Sheffield\\"}"
            >
              Sheffield
            </a>
          </div>
          <div
            class="city-link-container"
          >
            <a
              class="city-link"
              href="/?query={\\"city\\":\\"Liverpool\\"}"
            >
              Liverpool
            </a>
          </div>
          <div
            class="sort-city-title"
          >
            Sort By
          </div>
          <div
            class="sort-link-container"
          >
            <a
              class="sort-link"
              href="/?sort={\\"price\\":1}"
            >
              Price Ascending
            </a>
          </div>
          <div
            class="sort-link-container"
          >
            <a
              class="sort-link"
              href="/?sort={\\"price\\":-1}"
            >
              Price Descending
            </a>
          </div>
        </div>
      </div>
      <div
        class="property-card-container"
      />
    </div>
  </div>
</DocumentFragment>
`;
